src/Integration.Vsix/SonarAnalyzerManager.cs

```diff
@@ -37,52 +44,232 @@ public sealed class SonarAnalyzerManager
+                    // (i.e. getting here means the rule is enabled).
+                    return true;
+
+                case NewConnectedMode.SonarLintMode.Connected:
+                    return cachedQualityProfiles.GetValueOrDefault(Language.CSharp)
+                        ?.Rules
+                        .Select(x => x.Key)
+                        .Intersect(supportedDiagnostics.Select(d => d.Id))
+                        .Any()
+                        ?? HasAnyRuleInSonarWay(supportedDiagnostics);
+
+                default:
+                    Debug.Fail($"Unexpected SonarLintMode: {this.activeSolutionBoundTracker.CurrentConfiguration.Mode}");
+                    return false; // No rule will be enabled
+            }
+        }
+
+        private bool HasAnyRuleInSonarWay(IEnumerable<DiagnosticDescriptor> diagnostics) =>
+            diagnostics.Any(d => d.CustomTags.Contains(DiagnosticTagsHelper.SonarWayTag));
+
+        private void ReportDiagnostic(ReportingContext context)
+        {
+            if (this.activeSolutionBoundTracker.CurrentConfiguration.Mode == NewConnectedMode.SonarLintMode.LegacyConnected)
+            {
+                Debug.Fail($"The VSIX {nameof(SonarAnalysisContext.ReportDiagnosticAction)} delegate should not have been " +
+                    "called in this mode.");
+                return;
+            }
+
+            if (context.SyntaxTree == null ||
+                context.Diagnostic == null)
+            {
+                Debug.Fail("The 'ReportDiagnostic' delegate is not expected to be called with either a null syntax tree or a " +
+                    "null diagnostic.");
+                return;
+            }
 
-            return HasConflictingAnalyzerReference(projectAnalyzerStatus) ||
-                this.GetIsBoundWithoutAnalyzer(projectAnalyzerStatus);
+            // A DiagnosticAnalyzer can have multiple supported diagnostics and we decided to run the rule as long as at least
+            // one of the diagnostics is enabled. Therefore we need to filter the reported issues.
+            if (!HasAnyDiagnosticEnabled(new[] { context.Diagnostic.Descriptor }))
+            {
+                return;
+            }
+
+            // We are in the new connected mode and the issue is suppressed
+            if (this.activeSolutionBoundTracker.CurrentConfiguration.Mode == NewConnectedMode.SonarLintMode.Connected &&
+                !suppressionHandler.ShouldIssueBeReported(context.SyntaxTree, context.Diagnostic))
+            {
+                return;
+            }
+
+            context.ReportDiagnostic(GetConfiguredDiagnostic(context.Diagnostic));
+        }
+
+        private Diagnostic GetConfiguredDiagnostic(Diagnostic diagnostic)
+        {
+            if (this.activeSolutionBoundTracker.CurrentConfiguration.Mode == NewConnectedMode.SonarLintMode.Connected)
+            {
+                // TODO: (not on V1) Update the diagnostic based on the ruleset
```

Comment:
On a later iteration we will be able to change the rule severity (and maybe more), this will be the place to do so.